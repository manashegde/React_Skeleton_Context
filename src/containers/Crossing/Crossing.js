import React from "react";
import { Route, Switch } from "react-router-dom";
import Loadable from "react-loadable";
import NotFound from "components/NotFound";
import Loading from "components/Loading";
import ErrorBoundary from "components/ErrorBoundary";
import { routePaths } from "containers/Routes";
import "./Crossing.css";

const AsyncSearch = Loadable({
  loader: () => import("containers/Crossing/Search"),
  loading: Loading
});

const AsyncSpecialSearch = Loadable({
  loader: () => import("containers/Crossing/Search/SpecialSearch"),
  loading: Loading
});

const AsyncSamePassport = Loadable({
  loader: () => import("containers/Crossing/Search/SamePassport"),
  loading: Loading
});

const AsyncSearchResult = Loadable({
  loader: () => import("containers/Crossing/SearchResult"),
  loading: Loading
});

const AsyncTripInfo = Loadable({
  loader: () => import("containers/Crossing/TripInfo"),
  loading: Loading
});

const AsyncHitReport = Loadable({
  loader: () => import("containers/Crossing/HitReport"),
  loading: Loading
});

const AsyncMiscreantDetail = Loadable({
  loader: () => import("containers/Crossing/HitReport/MiscreantDetail"),
  loading: Loading
});

const AsyncWantedDetail = Loadable({
  loader: () => import("containers/Crossing/HitReport/WantedDetail"),
  loading: Loading
});

const AsyncAfisDetail = Loadable({
  loader: () => import("containers/Crossing/HitReport/AfisDetail"),
  loading: Loading
});

const AsyncAfisDetailEnrollment = Loadable({
  loader: () =>
    import("containers/Crossing/HitReport/AfisDetail/AfisDetailEnrollment"),
  loading: Loading
});

const AsyncFingerVerification = Loadable({
  loader: () => import("containers/Crossing/FingerVerification"),
  loading: Loading
});

class Crossing extends React.Component {
  render() {
    return (
      <ErrorBoundary>
        <div id="crossing">
          <Switch>
            <Route
              exact
              path={routePaths.crossing.search}
              component={AsyncSearch}
            />
            <Route
              exact
              path={routePaths.crossing.specialSearch}
              component={AsyncSpecialSearch}
            />
            <Route
              exact
              path={routePaths.crossing.samePassport}
              component={AsyncSamePassport}
            />
            <Route
              exact
              path={routePaths.crossing.tripInfo}
              component={AsyncTripInfo}
            />
            <Route
              exact
              path={routePaths.crossing.searchResult}
              component={AsyncSearchResult}
            />
            <Route
              exact
              path={routePaths.crossing.fingerVerification}
              component={AsyncFingerVerification}
            />
            <Route
              exact
              path={routePaths.crossing.hitReport}
              component={AsyncHitReport}
            />
            <Route
              exact
              path={routePaths.crossing.miscreantDetail}
              component={AsyncMiscreantDetail}
            />
            <Route
              exact
              path={routePaths.crossing.wantedDetail}
              component={AsyncWantedDetail}
            />
            <Route
              exact
              path={routePaths.crossing.afisDetail}
              component={AsyncAfisDetail}
            />
            <Route
              exact
              path={routePaths.crossing.afisDetailEnrollment}
              component={AsyncAfisDetailEnrollment}
            />
            <Route component={NotFound} />
          </Switch>
        </div>
      </ErrorBoundary>
    );
  }
}

export default Crossing;
