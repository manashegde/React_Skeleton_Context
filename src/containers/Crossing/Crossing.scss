@import "styles/colors.scss";

#crossing {
  // display: flex;
  // flex-direction: column;
  // align-items: center;
  height: 100%;
  overflow: auto;
  .label,
  .subtitle {
    // color: $white;
  }
  // padding: 2rem;
  // .form-box {
  //   background: $gray-gradient;
  //   padding: 2rem 4rem 4rem 4rem;
  //   .table thead th {
  //     color: #7a7a7a;
  //   }
  //   .select,
  //   select {
  //     width: 100%;
  //   }
  //   .subtitle {
  //     color: $white;
  //     text-align: center;
  //   }
  //   label {
  //     font-weight: 400;
  //     color: $white;
  //   }
  //   .table {
  //     thead th {
  //       background-color: rgba(85, 111, 127, 0.95);
  //       border: none;
  //       color: $white;
  //       position: sticky;
  //       top: -1px;
  //     }
  //   }
  // }
}
@media only screen and (max-width: 565px) {
  #crossing {
    align-items: stretch;
  }
}
