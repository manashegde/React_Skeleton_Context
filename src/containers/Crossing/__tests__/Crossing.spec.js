import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Crossing from "containers/Crossing";

const props = {
  match: { url: "" }
};
describe("Crossing", () => {
  it("should match snapshot", () => {
    const tree = toJson(shallow(<Crossing {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
