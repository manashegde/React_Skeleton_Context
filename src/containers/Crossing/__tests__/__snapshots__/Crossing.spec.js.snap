// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crossing should match snapshot 1`] = `
<CrossingContextProvider>
  <ErrorBoundary>
    <div
      id="crossing"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/search"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/tripinfo"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/searchresult"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/hitreport"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </div>
  </ErrorBoundary>
</CrossingContextProvider>
`;
